identity/client/src/components/form/JSONEditor.tsx

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+
+import Editor, { useMonaco } from "@monaco-editor/react";
+import { Button, FormLabel, Stack } from "@carbon/react";
+import { observer } from "mobx-react-lite";
+import {
+  ComponentProps,
+  FC,
+  useEffect,
+  useLayoutEffect,
+  useRef,
+  useState,
+} from "react";
+import { beautify as beautifyJSON } from "src/utility/components/editor/jsonUtils.ts";
+import { options } from "src/utility/components/editor/options.ts";
+import useTranslate from "src/utility/localization";
+import { Copy, Edit } from "@carbon/react/icons";
+import Flex from "src/components/layout/Flex.tsx";
+import { spacing03, supportError } from "@carbon/elements";
+import { useNotifications } from "src/components/notifications";
+
+type EditorFirstParam = Parameters<
+  NonNullable<ComponentProps<typeof JSONEditor>["onMount"]>
+>[0];
+
+type JSONEditorProps = {
+  value: string;
+  onChange?: (value: string) => void;
+  readOnly?: boolean;
+  onValidate?: (isValid: boolean) => void;
+  onMount?: (editor: {
+    showMarkers: () => void;
+    hideMarkers: () => void;
+  }) => void;
+};
+
+const JSONEditor: FC<JSONEditorProps> = observer(
```

Comment:
:question: I'm not familiar with the mobx library. Can you shortly explain, what it's used for in this case? Just making sure it's necessary to add that dependency.